### PR TITLE
DT-6175 part 4

### PR DESCRIPTION
**Pull request by vesameskanen**
- Refactor Itinerary page jsx rendering by using shared code blocks, don't repeat same  code again and again
- Remove duplicate query
- Shared wrapper for detail view
- Correct spinners while loading
- Correct notifiers and error messages
- Refactor planParamUtil and settings management
- Simplify
- Remove annoying ticket name comments from code

### CODE REVIEW

File: app/component/ItineraryPage.js [hunk 1]
```diff
@@ -259,8 +256,6 @@ class ItineraryPage extends React.Component {
           result.bikeAndPublicPlan?.itineraries,
         );
 
-        // show 6 bike + transit itineraries, preferably 3 of both kind.
-        // If there is not enough of a kind, take more from the other kind
```
Comment (optionsome) [inline app/component/ItineraryPage.js]: Why is this removed now?

File: app/component/ItineraryPage.js [hunk 2]
```diff
@@ -472,12 +465,10 @@ class ItineraryPage extends React.Component {
       date: earliestArrivalTime.format('YYYY-MM-DD'),
       time: earliestArrivalTime.format('HH:mm'),
     };
-    this.setState({
-      loadingMore: reversed ? spinnerPosition.bottom : spinnerPosition.top,
-    });
-    this.showScreenreaderLoadingAlert();
+    this.setState({ loadingMore: reversed ? 'bottom' : 'top' });
```
Comment (optionsome) [inline app/component/ItineraryPage.js]: Why do we go back to hardcoding strings here?

File: app/component/ItineraryPage.js [hunk 3]
```diff
@@ -623,16 +614,13 @@ class ItineraryPage extends React.Component {
...
+      // in detail view
+      // If itinerary is not found in detail view, go back to summary view
       if (
-        !isEqual(
-          // if intermediate places change
-          getIntermediatePlaces(prevProps.match.location.query),
-          viaPoints,
-        ) || // detail view has no itineraries to show
-        (!state.loadingAlt && !this.mapHashToPlan(hash)?.itineraries?.length)
+        // loadingAlt is first undefined, then true and finally false
```
Comment (optionsome) [inline app/component/ItineraryPage.js:635]: Maybe it's better to define some three state enum instead of having this fuzzy logic.

File: app/component/ItineraryPage.js [hunk 4]
```diff
@@ -1149,305 +1082,158 @@ class ItineraryPage extends React.Component {
...
+
+    if (desktop) {
+      const title = (
+        <FormattedMessage
+          id={detailView ? 'itinerary-page.title' : 'summary-page.title'}
+          defaultMessage="Itinerary suggestions"
+        />
+      );
+      const bckBtnFallback =
+        detailView || altTransitHash.includes(hash) ? 'pop' : undefined;
```
Comment (optionsome) [inline app/component/ItineraryPage.js:1234]: this seems a bit weird

File: app/component/ItineraryPageUtils.js [hunk 5]
```diff
@@ -358,6 +360,8 @@ export function checkDayNight(iconId, timem, lat, lon) {
   return iconId;
 }
 
+const streetLegModes = ['WALK', 'BICYCLE', 'CAR', 'SCOOTER'];
```
Comment (optionsome) [inline app/component/ItineraryPageUtils.js]: Spelling of these static helper variables could be STREET_LEG_MODES but not that important.

File: app/component/ItineraryPage.js (hunk 1 above)
Comment (vesameskanen) [reply app/component/ItineraryPage.js]: Strange. Must be some git merge thing. Restored now.

File: app/component/ItineraryPage.js (hunk 2 above)
Comment (vesameskanen) [reply app/component/ItineraryPage.js]: Same as above. A bit worrying. Restored.

File: app/component/ItineraryPage.js (hunk 4 above)
Comment (vesameskanen) [reply app/component/ItineraryPage.js:1234]: added a comment

File: app/component/ItineraryPage.js (hunk 3 above)
Comment (vesameskanen) [reply app/component/ItineraryPage.js:635]: Agreed and done.

File: app/component/ItineraryPageUtils.js (hunk 5 above)
Comment (vesameskanen) [reply app/component/ItineraryPageUtils.js]: done
